opencog/atoms/core/Variables.cc

```diff
@@ -539,12 +539,12 @@ Handle FreeVariables::substitute_scoped(const Handle& term,
 		if (GLOB_NODE == h->get_type())
 		{
 			Handle glst(substitute_scoped(h, args, silent, index_map, quotation));
-			if (glst->is_node())
-				return glst;
-
 			changed = true;
-			for (const Handle& gl : glst->getOutgoingSet())
-				oset.emplace_back(gl);
+			if (glst->get_type() == LIST_LINK)
```

Comment:
I believe that this is unwrapping the results from `wrap_glb_with_list()`, is that right? If so, can you add a comment saying this?

Reply:
I will add this comment myself, in just a moment.